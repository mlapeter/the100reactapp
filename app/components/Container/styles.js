import EStyleSheet from "react-native-extended-stylesheet";
import { colors, fontSizes, fontStyles } from "../../styles";

export default EStyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "space-between",
    backgroundColor: "$bgColor",
    paddingHorizontal: 5,
    width: "100%",
    paddingVertical: 10,
    backgroundColor: colors.veryDarkGrey
  }
});
