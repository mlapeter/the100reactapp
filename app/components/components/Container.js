// @flow
import * as React from "react";
import {StyleSheet, View} from "react-native";

type ContainerProps = {
    children: React.Node
};

export default class Container extends React.PureComponent<ContainerProps> {

    render(): React.Node {
        return (
            <View style={styles.container}>{this.props.children}</View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1
    }
});
