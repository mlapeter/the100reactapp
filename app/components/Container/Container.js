import React, { Component } from "react";
import PropTypes from "prop-types";
import { View, SafeAreaView, Platform } from "react-native";

import styles from "./styles";

class Container extends Component {
  static propTypes = {
    children: PropTypes.any
  };

  render() {
    if (Platform.OS === "ios") {
      return (
        <SafeAreaView style={styles.container}>
          {this.props.children}
        </SafeAreaView>
      );
    } else {
      return <View style={styles.container}>{this.props.children}</View>;
    }
  }
}

export default Container;
