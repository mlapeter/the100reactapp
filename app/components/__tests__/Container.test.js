import React from "react";
import { View } from "react-native";
import renderer from "react-test-renderer";

import { Container, styles } from "../Container";

it("exports a styles object", () => {
  expect(typeof styles).toBe("object");
});

it("renders successfully without children", () => {
  const rendered = renderer.create(<Container />).toJSON();
  console.log(rendered);
  expect(rendered).toBeTruthy();
});

it("renders children props", () => {
  const rendered = renderer
    .create(
      <Container>
        <View />
      </Container>
    )
    .toJSON();
  expect(rendered).toMatchSnapshot();
});
