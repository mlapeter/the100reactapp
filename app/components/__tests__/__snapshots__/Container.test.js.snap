// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children props 1`] = `
<RCTSafeAreaView>
  <View />
</RCTSafeAreaView>
`;
